client/app/main/visualization/visualization.controller.js

```diff
@@ -2,97 +2,115 @@
+        plotBorderWidth: null,
+        plotShadow: false,
+        type: 'pie'
+      },
       title: {
-        text: $scope.truancyType
+        text: 'ARCA vs CA'
       },
-      allowDecimals: $scope.truancyType === 'Average'
-    },
-    title: {
-      text: $scope.statType + ' ' + $scope.truancyType + ' ' + 'Trends'
+      tooltip: {
+        pointFormat: '{series.name}: <b>{point.percentage:.1f}%</b>'
+      },
+      plotOptions: {
+        pie: {
+          allowPointSelect: true,
+          cursor: 'pointer',
+          dataLabels: {
+            enabled: false,
+            format: '<b>{point.name}</b>: {point.percentage:.1f} %',
+            style: {
+                color: (Highcharts.theme && Highcharts.theme.contrastTextColor) || 'black'
+            }
+          },
+          showInLegend: true
+        }
+      }
     },
-    loading: false,
     func: function(chart) {
       $scope.chartObj = chart;
     }
   };
 
-  Visualization.schools().$promise.then(function(results) {
-    $scope.Cumulative = {
-      Absences: [],
-      Tardies: []
-    };
-    $scope.Average = {
-      Absences: [],
-      Tardies: []
-    };
-    _.forEach(results, function(school) {
-      var cumulativeAbsencesData = [];
-      var averageAbsencesData = [];
-      var cumulativeTardiesData = [];
-      var averageTardiesData = [];
-
-      school.data = _.sortBy(school.data, function(record) {
-        return record.date;
-      });
-
-      _.forEach(school.data, function(record) {
-        var dt = new Date(record.date);
-        var utcDt = Date.UTC(dt.getFullYear(), dt.getMonth(), dt.getDate());
-        cumulativeAbsencesData.push([utcDt, record.absences]);
-        averageAbsencesData.push([utcDt, record.absences / record.count]);
-        cumulativeTardiesData.push([utcDt, record.tardies]);
-        averageTardiesData.push([utcDt, record.tardies / record.count]);
-      });
-
-      $scope.Cumulative.Absences.push({
-        name: school._id.name,
-        data: cumulativeAbsencesData
-      });
-
-      $scope.Average.Absences.push({
-        name: school._id.name,
-        data: averageAbsencesData
-      });
-
-      $scope.Cumulative.Tardies.push({
-        name: school._id.name,
-        data: cumulativeTardiesData
-      });
-
-      $scope.Average.Tardies.push({
-        name: school._id.name,
-        data: averageTardiesData
-      });
-
-      $scope.chartConfig.series = $scope[$scope.statType][$scope.truancyType];
+  Visualization.arcaCa().$promise.then(function(results) {
+    function counter(rec, isCfa) {
+      var key = isCfa ? 'cfa' : 'nonCfa';
+      if(rec.arca) { 
+        $scope.data[key].arca++; 
+      } else if(rec.ca) { 
+        $scope.data[key].ca++; 
+      } else { 
+        $scope.data[key].other++; 
+      }
+    }    
+    _.forEach(results, function(record, key) {
+      if(record.student.cfa) { 
+        counter(record, true);
+      } else {
+        counter(record, false);
+      }
     });
+    
```

Comment:
So i took a closer look and it seems like this can't be done completely in the aggregation pipeline because mLab's version of mongo does not have $lookup which is introduced in mongo 3.2. $lookup is " equivalent to a left outer equi-join." https://www.mongodb.com/blog/post/joins-and-other-aggregation-enhancements-coming-in-mongodb-3-2-part-1-of-3-introduction